src/bitcoin-cli.cpp

```diff
@@ -478,7 +478,7 @@ class NetinfoRequestHandler : public BaseRequestHandler
             const int8_t network_id{NetworkStringToId(network)};
             if (network_id == UNKNOWN_NETWORK) continue;
             const bool is_outbound{!peer["inbound"].get_bool()};
-            const bool is_tx_relay{peer["relaytxes"].isNull() ? true : peer["relaytxes"].get_bool()};
```

Comment:
`bitcoin-cli` could be being used with an older version of `bitcoind`, no? I think `get_bool` treats null as `false`, but I think `getInt` will throw. Probably not an issue if only unreleased versions of `bitcoind` treated these values as optional though.

Reply:
I don't think we support arbitrary version combinations of bitcoin-cli and the rpc server? The only official supported combination is the one where both are built from the exact same commit hash.

Reply:
Right, for the moment the newly frequently missing fields are not in a release.

`get_bool` and `getInt` both throw on null with `error: JSON value of type null is not of expected type {bool|number}`, which can be seen with a snippet like this added to a CLI or RPC.

```cpp
const UniValue x = UniValue();
assert (x.isNull());
// auto y = x.get_bool();        // uncomment to throw
// auto z = x.getInt<int64_t>(); // uncomment to throw
```

Reply:
> `bitcoin-cli` could be being used with an older version of `bitcoind`, no?

Circling back to this, as I have heard in the past from people using a recent -netinfo with nodes that have been up since an earlier release. Updated the minimum server check in -netinfo from v21 to v22, the last version that should be backward compatible (due to bitcoin core adding the addr_* fields in v22 used in -netinfo) and dropped only the relaytxes null check and not the addr_* ones to maintain that backward compatibilty.